### PR TITLE
switch package source from GitHub to PyPI

### PR DESCRIPTION
**Pull request by grlee77**
I'm not sure why we had been using GitHub rather than PyPI in this recipe.  I switched to PyPI here just to be more in-line with the preferred conda-forge style

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 
 package:
-  name: pywavelets
+  name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://github.com/PyWavelets/pywt/archive/v{{ version }}.tar.gz
-  sha256: b152d4b551e3b454217c907e8722f7d8dc0adf104e6b3a8657d07e4020a990f2
+  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipe/meta.yaml]: You don't need the `fn`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: a12c7a6258c0015d2c75d88b87393ee015494551f049009e8b63eafed2d78efc
 
 build:
-  number: 0
+  number: 1
   script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Let's take this opportunity and update this to `script: "{{ PYTHON }} -m pip install . --no-deps -vv"`

File: recipe/meta.yaml (hunk 1 above)
Comment (grlee77) [reply recipe/meta.yaml]: Thanks. I have made the suggested change

**Comment by grlee77** [comment]
This seems fine on CI, but when I tried it locally on an OS X laptop I was unable to build the recipe. Has something changed in how local builds need to be performed?  I tried:

`conda build -c conda-forge -m .ci_support/osx_python3.6.yaml recipe` (either with or without the `m .ci_support/osx_python3.6.yaml` included)

and get
```
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform osx-64: {'clang_osx-64=4'}
```
I think this used to work, but perhaps the main OS X version has just become to old? (The system has OS X Yosemite 10.10.5.  I think 10.9 used to be the minimum version supported, but am not sure if that is still the case)

**Comment by ocefpaf** [comment]
Building things outside of the conda-forge infrastructure can be challenging. Check you `.condarc`, maybe you are missing `defaults`, which provides `clang_osx-64`.

**Comment by grlee77** [comment]
Thanks @ocefpaf. The culprit in `clang_osx-64` failing to install were some outdated `libcxx` and `libcxxabi` packages that I had to force remove and reinstall.

**Pull request merged**